app/src/main/java/com/udacity/sandwichclub/utils/JsonUtils.java

```diff
@@ -2,9 +2,42 @@
 
 import com.udacity.sandwichclub.model.Sandwich;
 
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
 public class JsonUtils {
 
     public static Sandwich parseSandwichJson(String json) {
-        return null;
+        try {
+            JSONObject sandwich = new JSONObject(json);
+            JSONObject name = sandwich.getJSONObject("name");
+            String mainName = name.getString("mainName");
+            List<String> alsoKnownAs = new ArrayList<>();
+            JSONArray akaArray = name.getJSONArray("alsoKnownAs");
+            for (int i = 0; i < akaArray.length(); i++) {
+                alsoKnownAs.add(akaArray.getString(i));
+            }
+            String placeOfOrigin = sandwich.getString("placeOfOrigin");
```

Comment:
Spacing between the different variables might be good for easier readability. You already do so in the `DetailActivity.populateUI` function, so would recommend making consistent across files